### DISCUSSION TITLE
Script runs beautifully, but can't download modified files

### DISCUSSION THREAD

**Discussion by pflaschner1970** [ricklamers/gpt-code-ui#29] [Q&A]
First, holy moly there is so much I want to use this for. Thank you.

Second, I get an error when I try to download a file from the UI. Here's the prompt and the response. Any help would be most appreciated.

Prompt: please strip html tags from this file

Response: I generated the following code:
from bs4 import BeautifulSoup

with open('filename.html') as file:
    soup = BeautifulSoup(file, 'html.parser')
    text = soup.get_text()

# print download link if needed
print(f"<a href='/download?file=filename.html'>Download file</a>")
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
Cell In[4], line 3
      1 from bs4 import BeautifulSoup
----> 3 with open('filename.html') as file:
      4     soup = BeautifulSoup(file, 'html.parser')
      5     text = soup.get_text()

File /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/IPython/core/interactiveshell.py:284, in _modified_open(file, *args, **kwargs)
    277 if file in {0, 1, 2}:
    278     raise ValueError(
    279         f"IPython won't let you open fd={file} by default "
    280         "as it is likely to crash IPython. If you know what you are doing, "
    281         "you can use builtins' open."
    282     )
--> 284 return io_open(file, *args, **kwargs)

FileNotFoundError: [Errno 2] No such file or directory: 'filename.html'

**Comment by ricklamers** [comment]
Love the enthusiasm! I feel the same way.

I think your problem can be broken down into two parts: 1) opening your html file 2) getting it to output the download link.

1. Did you upload the file called “filename.html” first?
2. Try a different prompt that more explicitly asks it to print the download link for the file you want to download.

Here’s an example process that should work:

1. Upload a file “index.html”
2. Prompt: “extract the text content of this html file and store it to onlytext.txt, once completed print a download link for that txt file”
3. Click download link


Note that GPT-4 is a bit better at generating the download link correctly. Generating the download link is “taught” one-shot in the internal prompt and GPT-3.5 doesn’t always follow that instruction.

Feel free to re-open if this doesn’t work.

**Comment by pflaschner1970** [comment]
Right on, thanks Rick!
